cli/fathom_web/commands/vectorize.py

```diff
@@ -0,0 +1,312 @@
+                break
+            now_completed_samples = len([line for line in status_box.text.splitlines() if line.endswith(': vectorized')])
+            bar.update(now_completed_samples - completed_samples)
+            completed_samples = now_completed_samples
+
+    new_file = look_for_new_vector_file(downloads_dir, vector_files_before)
+    print(f'Vectors saved to {str(new_file)}')
+    return firefox
+
+
+def get_fathom_fox_uuid(firefox):
+    """
+    Try to get the internal UUID for FathomFox from `prefs.js`.
+
+    We use a loop to try multiple times because the `prefs.js` file needs a
+    little time to update before the fathom addon information appears. Five
+    seconds seems adequate since one second has always worked for me (Daniel).
+    """
+    for _ in range(5):
+        prefs = (pathlib.Path(firefox.capabilities.get('moz:profile')) / 'prefs.js').read_text().split(';')
+        uuids = next((line for line in prefs if 'extensions.webextensions.uuids' in line)).split(',')
+        try:
+            fathom_fox_uuid = next((line for line in uuids if 'fathomfox@mozilla.com' in line)).split('\\"')[3]
+            return fathom_fox_uuid
+        except StopIteration:
+            time.sleep(1)
+    else:
+        raise GracefulError('Could not find UUID for FathomFox. No entry in the pres.js file.')
+
+
+def vector_files_present(firefox):
+    """Return the number of vector-y files present in Firefox's download dir."""
+    download_dir = pathlib.Path(firefox.profile.default_preferences['browser.download.dir'])
+    vector_files = download_dir.glob('vector*.json')
+    return len(list(vector_files))
+
+
+def extract_error_from(status_text):
+    """
+    Look for errors in the vectorizer's status text.
+
+    If there is an error, raise an exception causing an ungraceful shutdown.
+    """
+    lines = status_text.splitlines()
+    for line in lines:
+        if 'failed:' in line:
+            return line
+    raise UngracefulError(f'There was a vectorizer error, but we could not find it in {status_text}')
+
+
+def look_for_new_vector_file(downloads_dir, vector_files_before):
+    """
+    Look for a new vector file in the downloads directory.
+
+    We use a loop to try multiple times because the file system needs a little
+    little time to update before the file appears. Five seconds seems adequate
+    since one second has always worked for me (Daniel).
+    """
+    for _ in range(5):
+        vector_files_after = set(downloads_dir.glob('vector*.json'))
+        try:
+            new_file = (vector_files_after - vector_files_before).pop()
+            return new_file
+        except KeyError:
+            time.sleep(1)
+    else:
+        raise GracefulError(f'Could not find vectors file in {downloads_dir.as_posix()}.\n{vector_files_before} were already present.')
+
+
+def teardown(firefox, firefox_pid, geckoview_pid, server, server_thread, graceful_shutdown):
+    """
+    Close Firefox and the HTTP server.
```

Comment:
Should we also restore changes to `fathom-trainees`? (`git checkout -- .`) I'm just thinking about resetting the state of things at the end of the command. I don't think this particular task is going to break anything necessarily, it just modifies that repo and doesn't undo that modification.

Reply:
That is a very good point. Do you think it is important enough to do in this version of the tool or is an enhancement issue adequate?

Reply:
Yeah I think you could just add a TODO comment. As long as the `shutil.copyfile` command handles the situation where the file already exists, which it must, since `fathom-trainees` already comes with a default `trainees.js` file.